package/src/utilities/add-vite-plugin.ts

```diff
@@ -15,17 +44,33 @@ import type { PluginOption } from "vite";
  * ```ts
  * addVitePlugin({
  * 		plugin,
+ * 		config,
+ * 		logger,
  * 		updateConfig
  * })
  * ```
  */
 export const addVitePlugin = ({
+	warnDuplicated = true,
 	plugin,
+	config,
+	logger,
 	updateConfig,
-}: {
-	plugin: PluginOption;
-	updateConfig: HookParameters<"astro:config:setup">["updateConfig"];
-}) => {
+}: WarnDuplicateOptions | NoWarnDuplicateOptions) => {
+	if (config && hasVitePlugin({ plugin, config })) {
+		if (warnDuplicated && logger) {
+			logger.warn(
+				`The Vite plugin "${
+					(plugin as Plugin).name
+				}" is already present in your Vite configuration, this plugin may not behave correctly.`,
+			);
+		} else {
+			incrementPluginName(plugin as Plugin);
```

Comment:
If I understand correctly, this allows to duplicate plugins properly right? Isn't it "dangerous"? I don't know what it does when you have 2 duplicated ids actually

Reply:
🤦‍♂️ Good point! I implemented this as a way of making the utils that depends on `addVitePlugin` safer without also needing a `logger` param. I could make this toggleable with another option, something like `incrementName: true`?

Reply:
I think I found a better compromise by moving the incrementing plugin name to the `addVirtualImports` utility. This means plugins won't be duplicated inside `addVitePlugin` and the  `addVirtualImports`/`addDevToolbarThing` utils are safe with incrementing plugins names